File: knowledge/fixtures/discord-node-guild-bot/repo/src/pins.js
const fetch = require("node-fetch");

const headers = { Authorization: `Bot ${process.env.DISCORD_TOKEN}` };

async function listPins(channelId) {
  const res = await fetch(`https://discord.com/api/v8/channels/${channelId}/pins`, { headers });
  return res.json();
}

async function pin(channelId, messageId) {
  return fetch(`https://discord.com/api/v8/channels/${channelId}/pins/${messageId}`, { method: "PUT", headers });
}

module.exports = { listPins, pin };

File: knowledge/fixtures/discord-negative-current-routes/repo/archiver/pins.py
import os

import requests

API = "https://discord.com/api/v10"


def pinned_messages(channel_id: str):
    url = f"https://discord.com/api/v10/channels/{channel_id}/messages/pins"
    headers = {"Authorization": f"Bot {os.environ['DISCORD_TOKEN']}"}
    r = requests.get(url, params={"limit": 50}, headers=headers)
    r.raise_for_status()
    return r.json()


def guild(guild_id: str):
    return requests.get(f"https://discord.com/api/v10/guilds/{guild_id}", headers={"Authorization": f"Bot {os.environ['DISCORD_TOKEN']}"}).json()
